README.md

```diff
@@ -202,6 +202,37 @@ MainScope mainScope = new MainScopeImpl();
 ChildScope childScope = mainScope.child();
 ```
 
+## Dependency Resolution and Exposure
+
+Motif requires explicit dependency declarations to properly resolve and inject dependencies across scopes. When a parent scope needs to provide a dependency to a child scope, the dependency must be explicitly exposed using the `@Expose` annotation.
```

Comment:
This is valuable documentation, but can we add the other way to pass dependency to children as well? 

```kotlin
interface ParentScope {
   fun childScope(api: Api): ChildScope // Api is exposed to children through a scope factory method parameter.

   @motif.Objects
    abstract class Objects {
        fun provideImpl(): Impl {
            return Impl()
        }

        fun provideApi(impl: Impl): Api {
            return impl
        }
    }
}
```